### PR TITLE
leader: check client in GetLeader().

### PR DESCRIPTION
**Pull request by huachaohuang**
PD will panic if it try to `GetLeader()` before `s.client` is ready in startup.

### CODE REVIEW

**Comment by siddontang** [comment]
any test?

File: server/leader_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		CmdType: pdpb.CommandType_AllocId,
+		AllocId: &pdpb.AllocIdRequest{},
+	}
+
+	for {
+		select {
+		case <-s.done:
+			return
+		default:
+			// We don't need to check the response and errror,
```
Comment (siddontang) [inline server/leader_test.go]: error

**Comment by siddontang** [comment]
Rest LGTM

**Comment by huachaohuang** [comment]
PLTA @overvenus

**Comment by overvenus** [comment]
LGTM

**Pull request merged**